### PR TITLE
Refactor state test

### PR DESCRIPTION
**Pull request by ikanago**
- Define `ASSERT_STATE_NEAR` macro to test `QuantumState` easily
- Define factory function to create gate matrices in test
- Fix test that only checks absolute value of complex numbers; check the real and imaginary part

Thank you for reviewing the large diff 🙏

### CODE REVIEW

File: test/cppsim/test_circuit.cpp [hunk 1]
```diff
@@ -550,10 +523,7 @@ TEST(CircuitTest, CircuitOptimize) {
         qco.optimize(copy_circuit, block_size);
         circuit.update_quantum_state(&test_state);
         copy_circuit->update_quantum_state(&state);
-        // std::cout << circuit << std::endl << copy_circuit << std::endl;
-        for (UINT i = 0; i < dim; ++i)
-            ASSERT_NEAR(
-                abs(state.data_cpp()[i] - test_state.data_cpp()[i]), 0, eps);
+
```
Comment (forest1040) [inline test/cppsim/test_circuit.cpp:526]: ここにもASSERT_STATE_NEAR()が必要な気がします。

**Comment by forest1040** [comment]
`ASSERT_STATE_NEAR` and `make_xx` Very useful!! nice!

File: test/cppsim/test_circuit.cpp [hunk 2]
```diff
@@ -465,10 +448,7 @@ TEST(CircuitTest, CircuitOptimize) {
...
```
Comment (KowerKoint) [inline test/cppsim/test_circuit.cpp:451]: `ASSERT_STATE_NEAR` is also expected here.

File: test/cppsim/test_circuit.cpp [hunk 3]
```diff
@@ -493,10 +473,7 @@ TEST(CircuitTest, CircuitOptimize) {
...
```
Comment (KowerKoint) [inline test/cppsim/test_circuit.cpp:476]: `ASSERT_STATE_NEAR` is also expected here.

**Review by KowerKoint — COMMENTED** [review]
Thank you!
The content is much neater.

**Comment by ikanago** [comment]
Reflected your reviews.

**Review by forest1040 — APPROVED** [review]
👍

File: .github/workflows/ci_ubuntu.yml [hunk 4]
```diff
@@ -80,8 +80,8 @@ jobs:
       - name: Test in Ubuntu
         run: |
           cd ./build
-          make coverage
-          make pythontest
+          make coverage -j
+          make pythontest -j
```
Comment (kotamanegi) [inline .github/workflows/ci_ubuntu.yml:84]: concurrency rate should be limited to number of cpu cores.

```suggestion
          make coverage -j $(nproc)
          make pythontest -j $(nproc)
```

**Review by kotamanegi — APPROVED** [review]
LGTM:+1:
I've filled oversights in test_gate.cpp.

**Review by KowerKoint — APPROVED** [review]
LGTM

**Pull request merged**